### PR TITLE
xcute: Allow to excute tasks across all the platform

### PR DESCRIPTION
**Pull request by AymericDu**
##### SUMMARY

Allow to excute tasks across all the platform

1. The user creates a new job
2. The job dispatches all tasks to the workers
3. The worker processes the tasks

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- `xcute`
- `event-agent`
- `consience`
- Admin CLI

##### SDS VERSION

```
openio 5.2.4.dev126
```

### CODE REVIEW

File: oio/xcute/orchestrator.py [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+                sent = self.dispatch_tasks(
+                    beanstalkd_workers,
+                    job_id, job_type, job_config, tasks)
+                if sent:
+                    job_status, exc = self.handle_backend_errors(
+                        self.backend.update_tasks_sent, job_id, tasks.keys())
+                    tasks.clear()
+                    if exc is not None:
+                        self.logger.warn(
+                            '[job_id=%s] Sent tasks has not been updated: %s',
```
Comment (fvennetier) [inline oio/xcute/orchestrator.py]: Sent tasks counter?

File: oio/xcute/orchestrator.py [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+                    if exc is None:
+                        if job_status == 'FINISHED':
+                            self.logger.info('Job %s is finished', job_id)
+
+                        self.logger.info(
+                            'Finished dispatching job (job_id=%s)', job_id)
+                        return
+                    else:
+                        self.logger.warn(
+                            '[job_id=%s] Last sent tasks has not been '
```
Comment (fvennetier) [inline oio/xcute/orchestrator.py]: Sent tasks counter?

File: oio/xcute/orchestrator.py [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+                self.logger.warn(
+                    '[job_id=%s] Job has not been freed: %s',
+                    job_id, exc)
+        except Exception as exc:
+            self.logger.exception(
+                '[job_id=%s] Failed generating task list: %s', job_id, exc)
+            _, exc = self.handle_backend_errors(self.backend.fail, job_id)
+            if exc is not None:
+                self.logger.warn(
+                    '[job_id=%s] Failure has not been updated: %s',
```
Comment (fvennetier) [inline oio/xcute/orchestrator.py]: Failure counter?

File: oio/cli/admin/xcute/rawx.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        job_info = self.xcute.job_create(
+            RawxRebuildJob.JOB_TYPE, job_config=job_config)
+        return zip(*sorted(
+            flat_dict_from_dict(parsed_args, job_info).items()))
+
+
+class RawxDecommission(SingleServiceCommandMixin, ShowOne):
+    """
+    Decommission the specified service.
+    All chunks will be moved on the others services.
```
Comment (fvennetier) [inline oio/cli/admin/xcute/rawx.py]: All chunks matching the size constraints.

File: oio/cli/admin/xcute/job.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+    def take_action(self, parsed_args):
+        self.logger.debug('take_action(%s)', parsed_args)
+
+        return self.columns, self._take_action(parsed_args)
+
+
+class JobShow(XcuteCommand, ShowOne):
+    """
+    Get all informations about the job
```
Comment (fvennetier) [inline oio/cli/admin/xcute/job.py]: all information (without 's')

File: oio/cli/admin/xcute/job.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+    def take_action(self, parsed_args):
+        self.logger.debug('take_action(%s)', parsed_args)
+
+        return self.columns, self._take_action(parsed_args)
+
+
+class JobDelete(XcuteCommand, Lister):
+    """
+    Delete all informations about the jobs
```
Comment (fvennetier) [inline oio/cli/admin/xcute/job.py]: all information (without the 's')

File: oio/xcute/orchestrator.py (hunk 1 above)
Comment (AymericDu) [reply oio/xcute/orchestrator.py]: It's not just the counter that hasn't been updated. We also keep all the IDs of the sent tasks (until they are processed).

File: oio/xcute/orchestrator.py (hunk 2 above)
Comment (AymericDu) [reply oio/xcute/orchestrator.py]: It's not just the counter that hasn't been updated. We also keep all the IDs of the sent tasks (until they are processed).

File: oio/xcute/orchestrator.py (hunk 3 above)
Comment (AymericDu) [reply oio/xcute/orchestrator.py]: For this case, there is no counter. It's all the job that fails. Therefore, we pass the status on to `FAILED`.

File: oio/xcute/client.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        """Refresh xcute service endpoint."""
+        addr = self._get_xcute_addr(**kwargs)
+        self.endpoint = '/'. join(("http:/", addr, "v1.0/xcute"))
+        if not now:
+            now = time.time()
+        self._last_refresh = now
+
+    def _maybe_refresh_endpoint(self, **kwargs):
+        """Refresh xcute service endpoint if delay has been reached."""
+        if self._refresh_delay >= 0.0 or not self.endpoint:
```
[the remark this answers is not in the document]
Comment (AymericDu) [reply oio/xcute/client.py]: You can already do this by choosing a very large value for `refresh_delay`.

**Pull request merged**